deob-bytecode/src/main/kotlin/org/openrs2/deob/bytecode/transform/StringDecryptionTransformer.kt

```diff
@@ -0,0 +1,183 @@
+                                if (insn.opcode != Opcodes.AALOAD) continue
+
+                                val push = insn.previous
+                                if (push.previous.opcode != Opcodes.GETSTATIC) continue
+
+                                val getstatic = push.previous as FieldInsnNode
+                                if (
+                                    getstatic.owner != clazz.name ||
+                                    getstatic.name != field.name ||
+                                    getstatic.desc != field.desc
+                                ) {
+                                    continue
+                                }
+
+                                val index = push.intConstant ?: continue
+                                val str = strings[index]
+                                val ldc = LdcInsnNode(str)
+
+                                method.instructions.remove(push)
+                                method.instructions.remove(insn)
+                                method.instructions.set(getstatic, ldc)
+
+                                stringsInlined++
+                            }
+                        } else if (field.desc == "Ljava/lang/String;") {
+                            // one encrypted string in the class so no array is produced
+                            for (insn in method.instructions) {
+                                if (insn.opcode != Opcodes.GETSTATIC) continue
+
+                                val getstatic = insn as FieldInsnNode
+                                if (
+                                    getstatic.owner != clazz.name ||
+                                    getstatic.name != field.name ||
+                                    getstatic.desc != field.desc
+                                ) {
+                                    continue
+                                }
+
+                                val str = strings[0]
+                                val ldc = LdcInsnNode(str)
+
+                                method.instructions.set(getstatic, ldc)
+
+                                stringsInlined++
+                            }
+                        }
+                    }
+
+                    // step 5: cleanup
+                    clazz.fields.remove(field)
+                    clinit.instructions.deleteExpression(fieldInit!!)
+                }
+
+                // step 5: cleanup
+                clazz.methods.remove(inner)
+                clazz.methods.remove(outer)
+
+                classesDecrypted++
+            }
+        }
+
+        if (stringsDecrypted > 0 || classesDecrypted > 0 || stringsInlined > 0) {
+            logger.info { "Decrypted $stringsDecrypted strings across $classesDecrypted classes" }
+            logger.info { "Inlined $stringsInlined string references" }
+        }
+    }
+
+    private fun decryptString(str: String, xorLookup: ArrayList<Number>): String {
```

Comment:
Use `List` here, rather than `ArrayList`. This means it doesn't matter which actual implementation of the underlying interface is used.